### DISCUSSION TITLE
How can I get a list of global variables from lua?

### DISCUSSION THREAD

**Discussion by vitaly** [neovim/neovim#24177] [General]
I can't iterate over `vim.g` as it has a metatable. And I can't call `vim.api.nvim_call_function('keys', ...)` as can't pass it the internal  dictionary (if I could, I'd just iterate over it anyway).

**Comment by clason** [comment]
`vim.cmd.echo('g:')`

**Reply by vitaly** [reply]
well, duh! this only prints it. I need to get it into lua for processing.

**Comment by zeertzjq** [comment]
`vim.api.nvim_eval('g:')`
